### PR TITLE
#166841614 Configure migrations with the current codebase

### PR DESCRIPTION
**Pull request by nkpremices**
#### What does this PR do?
Have migrations working with the current codebase 

#### Description of Task to be completed?
* Setup migrations and ensure develop branch is properly integrated with migrations
* Document the database setup process on README

#### How should this be manually tested?
* Clone the repo 
* Make sure you have all the environment variables in the ```dotenv```
* Run the command ```npm run setup:test``` and see the logs

#### Screenshots (if appropriate)
 
Screen Shot 2019-06-21 at 21 50 35

### CODE REVIEW

**Review by Ramadhan0 — APPROVED** [review]
looks fine

**Comment by MCFrank16** [comment]
You are awesome man!!!!

**Review by MCFrank16 — APPROVED** [review]

**Comment by truestbyheart** [comment]
Please add a header before adding the screenshots
Refer to the engineering guide

File: README.md [hunk 1]
```diff
@@ -412,3 +411,26 @@ No additional parameters required
...
 
 `GET /api/tags`
+
+## Getting started
+
+### 1. Installation steps
+
+- Clone the repo using
+
+`https://github.com/andela/savage_rangers-ah.git`
```
Comment (MemunaHaruna) [inline README.md]: @nkpremices can we change this to:

```
Clone the repo using:

git clone https://github.com/andela/savage_rangers-ah.git

```

File: README.md [hunk 2]
```diff
@@ -412,3 +411,26 @@ No additional parameters required
...
+
+- Install dependecies by runing
```
Comment (MemunaHaruna) [inline README.md]: @nkpremices small typo here: `dependencies`

File: src/api/migrations/20190620192200-create-user.js [hunk 3]
```diff
@@ -0,0 +1,70 @@
+export default {
+  up: (queryInterface, Sequelize) => queryInterface.createTable('Users', {
+    id: {
+      allowNull: false,
+      autoIncrement: true,
+      primaryKey: true,
+      type: Sequelize.INTEGER
+    },
+    username: {
```
Comment (MemunaHaruna) [inline src/api/migrations/20190620192200-create-user.js:9]: @nkpremices username and email should be unique.

File: src/api/migrations/20190620192200-create-user.js [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+      type: Sequelize.STRING
+    },
+    email: {
+      type: Sequelize.STRING
+    },
+    password: {
+      type: Sequelize.STRING
+    },
+    country: {
```
Comment (MemunaHaruna) [inline src/api/migrations/20190620192200-create-user.js:20]: @nkpremices are country,  bio, address, gender, profileImage and phoneNumber required fields? If they aren't, then we should allow them to be null.

Also, what do the fields `facebook` and `twitter` represent?

File: src/helpers/sequelize.initializer.js [hunk 5]
```diff
@@ -0,0 +1,3 @@
+import models from '../api/models';
+
+models.SequelizeMeta.truncate({ truncate: { cascade: false } });
```
Comment (MemunaHaruna) [inline src/helpers/sequelize.initializer.js:3]: @nkpremices what does this do? Do you have a link to where you found this syntax?

**Review by MemunaHaruna — CHANGES_REQUESTED** [review]

File: src/helpers/sequelize.initializer.js (hunk 5 above)
Comment (premlogic) [reply src/helpers/sequelize.initializer.js:3]: It is truncating the ```sequelizeMeta```  table.
Oh.. There was a time I had issues with ```Travis``` 
I added this file to resolve them by truncating that table. 
Now as I've fixed the issue by using a ```Travis database service``` I think we no longer need this file. I will just go and delete it.

File: src/api/migrations/20190620192200-create-user.js (hunk 3 above)
Comment (premlogic) [reply src/api/migrations/20190620192200-create-user.js:9]: Noted @MemunaHaruna

File: README.md (hunk 2 above)
Comment (premlogic) [reply README.md]: Oh!!! Thank you @MemunaHaruna

File: README.md (hunk 1 above)
Comment (premlogic) [reply README.md]: Understood @MemunaHaruna

File: src/api/migrations/20190620192200-create-user.js (hunk 4 above)
Comment (premlogic) [reply src/api/migrations/20190620192200-create-user.js:20]: Actually ```sequelize``` allows null by default. 
@alainburindi told me that the ```facebook``` and ```twitter```  fields will contain the links to social media profiles

Comment (MemunaHaruna) [reply src/api/migrations/20190620192200-create-user.js:20]: Thanks @nkpremices

**Review by MemunaHaruna — APPROVED** [review]

**Pull request merged**